integration/simulation/simulation.go

```diff
@@ -77,3 +79,21 @@ func (s *Simulation) Stop() {
 		}
 	}()
 }
+
+// Waits for the L2 nodes to be ready to process P2P messages.
+func (s *Simulation) waitForP2p() {
+	if s.ObscuroNodes[0].P2pAddress != placeholderAddress { // We check we aren't dealing with in-memory nodes.
```

Comment:
I'd suggest to use a local structure for the p2paddress.
Also to add a todo to remove this

Reply:
Are we sure we want to remove this? It seems reasonable to me, waiting for the nodes to be ready to kick things off. Otherwise, there's a risk that some of the nodes would miss a big part of the simulation as the node becomes more complex and takes longer to start.

Reply:
that's actually a thing we want to test as part of this comprehensive simulation. That nodes are able to catch up when started at arbitrary points.
We'll also have to add restarting and other stuff like that.

Reply:
Will add a todo.